src/raft.rs

```diff
@@ -2124,6 +2257,20 @@ impl<T: Storage> Raft<T> {
         }
     }
 
+    /// Takes the `Groups`.
```

Comment:
These APIs conflict with other parts. I guess what we only need is `fn set_group_id(&mut self, id: u64)`?